### PR TITLE
Add reporting hooks to DetailsAction

### LINKED ISSUE #704: Action to create/modify report for displayed host/service
I would suggest an action (in the "Problem handling" part of the host/service detail), which will create a report  with the 'Report', 'Filter' and title fields filled in already (or perhaps even modify it, if it exists).

### PR DESCRIPTION
**Pull request by jhoxhaa**
fixes #704

### CODE REVIEW

**Review by yhabteab — COMMENTED** [review]
I think here is not the appropriate repository for these hooks. Also, those hooks shouldn't be object extensions, but rather `HostActions` and `ServiceActions` and have to be provided by the reporting module.

cc @nilmerg

**Comment by nilmerg** [comment]
No, that's the correct location. Icinga DB Web provides the sla report functionality for reporting. Reporting has no idea what SLA means and can't provide actions for this kind of report.

But you're correct with the assumption that these should be actions. Though, I'd even question whether this should be a hook at all. `ObjectDetail::createActions` could do this instead.

**Comment by lippserd** [comment]
> Though, I'd even question whether this should be a hook at all.

The reports are also hooks. I think it makes sense to keep the non-core module functionality separated.

**Review by nilmerg — CHANGES_REQUESTED** [review]
Please let the hooks extend `Icinga\Module\Icingadb\Hook\HostActionsHook` and `Icinga\Module\Icingadb\Hook\ServiceActionsHook`.

File: library/Icingadb/ProvidedHook/CreateServiceSlaReport.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+use ipl\Web\Url;
+
+class CreateServiceSlaReport extends ServicesDetailExtensionHook
+{
+    public function getHtmlForObjects(Query $services): ValidHtml
+    {
+        $url = Url::fromPath('reporting/reports/new');
+        $filter = QueryString::render($this->getBaseFilter());
+        $url->addParams(['filter' => $filter, 'report' => 'service']);
+        $wrapper = Html::tag('div');
```
Comment (nilmerg) [inline library/Icingadb/ProvidedHook/CreateServiceSlaReport.php:19]: Please don't use a wrapper element. The other content also doesn't use one. Use a `HtmlDocument` as wrapper instead.

File: library/Icingadb/ProvidedHook/CreateHostSlaReport.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+use ipl\Web\Url;
+
+class CreateHostSlaReport extends HostsDetailExtensionHook
+{
+    public function getHtmlForObjects(Query $hosts): ValidHtml
+    {
+        $url = Url::fromPath('reporting/reports/new');
+        $filter = QueryString::render($this->getBaseFilter());
+        $url->addParams(['filter' => $filter, 'report' => 'host']);
+        $wrapper = Html::tag('div');
```
Comment (nilmerg) [inline library/Icingadb/ProvidedHook/CreateHostSlaReport.php:19]: Please don't use a wrapper element. The other content also doesn't use one. Use a `HtmlDocument` as wrapper instead.

File: library/Icingadb/ProvidedHook/CreateServiceSlaReport.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        $wrapper
+            ->add(Html::tag('h2', 'Reporting'))
+            ->add(Html::tag('a', ['href' => $url], 'Create Service SLA Report'));
```
Comment (nilmerg) [inline library/Icingadb/ProvidedHook/CreateServiceSlaReport.php:22]: Please use the `ipl\Web\Widget\Link` class and translate both labels with `ipl\I18n\Translation::translate()`

File: library/Icingadb/ProvidedHook/CreateHostSlaReport.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        $wrapper
+            ->add(Html::tag('h2', 'Reporting'))
+            ->add(Html::tag('a', ['href' => $url], 'Create Host SLA Report'));
```
Comment (nilmerg) [inline library/Icingadb/ProvidedHook/CreateHostSlaReport.php:22]: Please use the `ipl\Web\Widget\Link` class and translate both labels with `ipl\I18n\Translation::translate()`

File: run.php [hunk 5]
```diff
@@ -4,13 +4,24 @@
...
+
 $this->provideHook('ApplicationState');
 $this->provideHook('X509/Sni');
 $this->provideHook('health', 'IcingaHealth');
 $this->provideHook('health', 'RedisHealth');
 $this->provideHook('Reporting/Report', 'Reporting/HostSlaReport');
 $this->provideHook('Reporting/Report', 'Reporting/ServiceSlaReport');
 
+if (
+    $this::exists('reporting') && Icinga::app()->getModuleManager()->hasEnabled('reporting')
```
Comment (nilmerg) [inline run.php:18]: `exists` already checks the enabled state if I'm not mistaken :thinking:

File: run.php [hunk 6]
```diff
@@ -4,13 +4,24 @@
...
+    && Auth::getInstance()->hasPermission('reporting/reports')
```
Comment (nilmerg) [inline run.php:19]: Accessing the authentication at this stage (run.php runs very early) may cause issues, especially on routes that normally don't need authentication. Placing this here causes everything to require authentication as run.php runs always. Please check the permission in the hooks itself.

**Review by nilmerg — CHANGES_REQUESTED** [review]

**Review by nilmerg — CHANGES_REQUESTED** [review]
Please put always an empty line after closing a control structure.

e.g.:
```
    }

    return $result;
```

Otherwise it looks fine now.

**Review by nilmerg — APPROVED** [review]

**Pull request merged**